tests/unit-ext/test-ext-method-register.c

```diff
@@ -0,0 +1,244 @@
+    jerryx_set_properties (target_object, methods);
+
+  jerryx_property_entry_cleanup (methods, register_result);
+
+  TEST_ASSERT (register_result.registered == 1);
+  TEST_ASSERT (jerry_value_is_undefined (register_result.result));
+
+  jerry_value_t global_obj = jerry_get_global_object ();
+  jerryx_set_property_str (global_obj, "test", target_object);
+  jerry_release_value (target_object);
+  jerry_release_value (global_obj);
+
+  {
+    const char *test_A = "test.hello(33, 42, 2);";
+    jerry_value_t result = jerry_eval ((const jerry_char_t*) test_A, strlen (test_A), 0);
+    TEST_ASSERT (jerry_value_is_number (result));
+    TEST_ASSERT ((uint32_t) jerry_get_number_value (result) == 3u);
+    jerry_release_value (result);
+  }
+
+  {
+    const char *test_A = "test.hello();";
+    jerry_value_t result = jerry_eval ((const jerry_char_t*) test_A, strlen (test_A), 0);
+    TEST_ASSERT (jerry_value_is_number (result));
+    TEST_ASSERT ((uint32_t) jerry_get_number_value (result) == 0u);
+    jerry_release_value (result);
+  }
+
+  jerry_cleanup ();
+} /* test_simple_registration */
+
+
+static void
+test_error_setvalue (void)
+{
+  jerry_init (JERRY_INIT_EMPTY);
+
+  const char *target_prop = "test_err";
+  jerry_value_t global_obj = jerry_get_global_object ();
+  freeze_property (global_obj, target_prop);
+
+  jerry_value_t target_object = jerry_create_object ();
+  jerry_value_t set_result =
+    jerryx_set_property_str (global_obj, target_prop, target_object);
+  TEST_ASSERT (jerry_value_is_error (set_result));
+
+  jerry_release_value (set_result);
+  jerry_release_value (target_object);
+  jerry_release_value (global_obj);
+
+  jerry_cleanup ();
+} /* test_error_setvalue */
+
+
+static void
+test_error_single_function (void)
+{
+  jerry_init (JERRY_INIT_EMPTY);
+
+  const char *target_prop = "test_err";
+  jerry_value_t target_object = jerry_create_object ();
+  freeze_property (target_object, target_prop);
+
+  jerryx_property_entry methods[] = {
+    { target_prop, jerry_create_external_function (method_hello) }, // This registration should fail
+    { NULL, 0 },
+  };
+
+  jerryx_register_result register_result =
+    jerryx_set_properties (target_object, methods);
+
+  jerryx_property_entry_cleanup (methods, register_result);
+
+  TEST_ASSERT (register_result.registered == 0);
+  TEST_ASSERT (jerry_value_is_error (register_result.result));
+  jerry_release_value (register_result.result);
+  jerry_release_value (target_object);
+
+  jerry_cleanup ();
+} /* test_error_single_function */
+
+
+static void
+test_error_multiple_functions (void)
+{
+  jerry_init (JERRY_INIT_EMPTY);
+
+  const char *prop_ok = "prop_ok";
+  const char *prop_err = "prop_err";
+  const char *prop_not = "prop_not";
+  jerry_value_t target_object = jerry_create_object ();
+  freeze_property (target_object, prop_err);
+
+  jerryx_property_entry methods[] = {
+    { prop_ok,  jerry_create_external_function (method_hello) }, // This registration is ok
```

Comment:
Could you update all the tests to use the recommended registration macros and broaden the tests with number, string, boolean registrations?